code/game/objects/structures/janitor.dm

```diff
@@ -1,13 +1,69 @@
 #define CART_HAS_MINIMUM_REAGENT_VOLUME !(reagents.total_volume < 1)
 
-/obj/structure/janitorialcart
-	name = "janitorial cart"
-	desc = "This is the alpha and omega of sanitation."
+/obj/structure/mopbucket
+	name = "mop bucket"
+	desc = "Fill it with water, but don't forget a mop!"
 	icon = 'icons/obj/janitor.dmi'
-	icon_state = "cart"
-	anchored = FALSE
+	icon_state = "mopbucket"
 	density = TRUE
 	var/amount_per_transfer_from_this = 5 //shit I dunno, adding this so syringes stop runtime erroring. --NeoFite
+	/// The icon used for the water overlay
+	var/water_icon = "mopbucket_water"
+
+/obj/structure/mopbucket/Initialize(mapload)
+	. = ..()
+	create_reagents(100, OPENCONTAINER)
+	register_context()
+
+/obj/structure/mopbucket/add_context(atom/source, list/context, obj/item/held_item, mob/living/user)
+	. = ..()
+
+	if(istype(held_item, /obj/item/mop))
+		context[SCREENTIP_CONTEXT_RMB] = "Wet [held_item]"
+		. = CONTEXTUAL_SCREENTIP_SET
+
+	if(istype(held_item, /obj/item/reagent_containers))
+		context[SCREENTIP_CONTEXT_LMB] = "Fill mop bucket"
+		. = CONTEXTUAL_SCREENTIP_SET
+
+	return . || NONE
+
+/obj/structure/mopbucket/attackby(obj/item/weapon, mob/user, params)
+	if(istype(weapon, /obj/item/reagent_containers))
+		update_appearance(UPDATE_OVERLAYS)
+		return FALSE // skip attack animation when refilling cart
+
+	return ..()
+
+/obj/structure/mopbucket/attackby_secondary(obj/item/weapon, mob/user, params)
+	var/obj/item/mop/mop = weapon
```

Comment:
Doesn't look like you use anything from /obj/item/mop, so you should probably remove this and check `if(istype(weapon, /obj/item/mop))`.